docs/advanced/smart_ptrs.rst

```diff
@@ -167,8 +154,70 @@ specialized:
 The above specialization informs pybind11 that the custom ``SmartPtr`` class
 provides ``.get()`` functionality via ``.getPointer()``.
 
+.. note::
+
+    The two noteworthy disadvantages mentioned under the ``std::shared_ptr``
+    section apply similarly to custom smart pointer holders, but there is no
+    established safe alternative in this case.
+
 .. seealso::
 
     The file :file:`tests/test_smart_ptr.cpp` contains a complete example
     that demonstrates how to work with custom reference-counting holder types
     in more detail.
+
+
+Be careful not to accidentally undermine automatic lifetime management
```

Comment:
Maybe I missed it, but is this a new section that has nothing to do with smart holder? Maybe move it to a separate PR.

Reply:
Yes, it's definitely odd here, but it's based on this existing part:

https://github.com/pybind/pybind11/blob/d8565ac7317b4838a08308170f5b9ba82d091546/docs/advanced/smart_ptrs.rst#L50-L113

I think it'll be best to move it only after this PR is merged, to not make this PR even more difficult to review.